driver/src/main/scala/api/DefaultCursor.scala

```diff
@@ -585,7 +585,8 @@ private[reactivemongo] object DefaultCursor {
         logger.trace(s"Asking for the next batch of $ntr documents on cursor #${reply.cursorID}, after ${nextOffset}: ${reqMaker.op}")
 
         def req = new ExpectingResponse(
-          requestMaker = reqMaker,
+          requestMaker =
+            reqMaker.copyWithChannelIdHint(Some(response.info.channelId)),
```

Comment:
Directly call the constructor rather than introducing a function used only once there

Reply:
In the `RequestMaker` class there are two private fields `kind` and `parentSTE`. I cannot build the same request from the previous using constructor due to this fields. 

Would you rather create a constructor taking the previous request and the field that need to be changed?